lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -90,23 +94,28 @@ class ClipboardAPIPasteStrategy implements PasteFromClipboardStrategy {
 /// Provides a fallback strategy for browsers which does not support ClipboardAPI.
 class ExecCommandCopyStrategy implements CopyToClipboardStrategy {
   @override
-  void setData(String text) {
-    // Copy content to clipboard with execCommand.
-    // See: https://developers.google.com/web/updates/2015/04/cut-and-copy-commands
-    final html.TextAreaElement tempTextArea = _appendTemporaryTextArea();
-    tempTextArea.value = text;
-    tempTextArea.focus();
-    tempTextArea.select();
-    try {
-      final bool result = html.document.execCommand('copy');
-      if (!result) {
-        print('copy is not successful');
+  Future<void> setData(String text) {
+    // In Flutter, platform messages are exchanged between threads so the
+    // messages and responses have to be exchanged asynchronously. We simulate
+    // that by adding a zero-length delay to the reply.
+    return Future.delayed(Duration.zero).then((_) {
```

Comment:
I think you can avoid the delay (and the new callback you are passing to `.then(...)`). What if we do `return Future.value(null);` in the end to make it return a future?

Reply:
done.

Reply:
In order to fix the 3rd comment and also to use try-catch at the same time, I returned a boolean.